chainio/clients/elcontracts/writer.go

```diff
@@ -41,37 +44,93 @@ type ELWriter interface {
-	signer            signer.Signer
-	logger            logging.Logger
-	ethClient         eth.EthClient
+	slasher                 slasher.ContractSlasherTransacts
+	delegationManager       delegationmanager.ContractDelegationManagerTransacts
+	strategyManager         strategymanager.ContractStrategyManagerTransacts
+	strategyManagerAddr     gethcommon.Address
+	blsPubkeyCompendium     blspubkeycompendium.ContractBLSPublicKeyCompendiumTransacts
+	blsPubkeyCompendiumAddr gethcommon.Address
+	elChainReader           ELReader
+	ethClient               eth.EthClient
+	signer                  signer.Signer
+	logger                  logging.Logger
 }
 
 var _ ELWriter = (*ELChainWriter)(nil)
 
 func NewELChainWriter(
-	elContractsClient clients.ELContractsClient,
+	slasher slasher.ContractSlasherTransacts,
+	delegationManager delegationmanager.ContractDelegationManagerTransacts,
+	strategyManager strategymanager.ContractStrategyManagerTransacts,
+	strategyManagerAddr gethcommon.Address,
+	blsPubkeyCompendium blspubkeycompendium.ContractBLSPublicKeyCompendiumTransacts,
+	blsPubkeyCompendiumAddr gethcommon.Address,
+	elChainReader ELReader,
 	ethClient eth.EthClient,
 	signer signer.Signer,
 	logger logging.Logger,
 	eigenMetrics metrics.Metrics,
 ) *ELChainWriter {
 	return &ELChainWriter{
-		elContractsClient: elContractsClient,
-		signer:            signer,
-		logger:            logger,
-		ethClient:         ethClient,
+		slasher:                 slasher,
+		delegationManager:       delegationManager,
+		strategyManager:         strategyManager,
+		strategyManagerAddr:     strategyManagerAddr,
+		blsPubkeyCompendium:     blsPubkeyCompendium,
+		blsPubkeyCompendiumAddr: blsPubkeyCompendiumAddr,
+		elChainReader:           elChainReader,
+		signer:                  signer,
+		logger:                  logger,
+		ethClient:               ethClient,
 	}
 }
 
+func BuildELChainWriter(
+	slasherAddr gethcommon.Address,
+	blsPubKeyCompendiumAddr gethcommon.Address,
+	ethClient eth.EthClient,
+	signer signer.Signer,
+	logger logging.Logger,
+	eigenMetrics metrics.Metrics,
+) (*ELChainWriter, error) {
+	elContractBindings, err := chainioutils.NewEigenlayerContractBindings(slasherAddr, blsPubKeyCompendiumAddr, ethClient, logger)
+	if err != nil {
+		return nil, err
+	}
+	elChainReader := NewELChainReader(
+		elContractBindings.Slasher,
+		elContractBindings.DelegationManager,
+		elContractBindings.StrategyManager,
+		elContractBindings.BlsPubkeyCompendium,
+		blsPubKeyCompendiumAddr,
+		logger,
+		ethClient,
+	)
+	return NewELChainWriter(
+		elContractBindings.Slasher,
+		elContractBindings.DelegationManager,
+		elContractBindings.StrategyManager,
+		elContractBindings.StrategyManagerAddr,
+		elContractBindings.BlsPubkeyCompendium,
+		blsPubKeyCompendiumAddr,
+		elChainReader,
+		ethClient,
+		signer,
+		logger,
+		eigenMetrics,
+	), nil
+}
+
+// TODO(madhur): do we really want to wait for txreceipts like this in these functions?
```

Comment:
let's keep it right now. This will change once we have a better tx manager design implemented in  #75

Reply:
Updated comment: [3d14cf3](https://github.com/Layr-Labs/eigensdk-go/pull/73/commits/3d14cf3a2662924a57d01e098c987bd33ce8efcf)